### DISCUSSION TITLE
Header is inconsistent after injecting the Fingerprints..

### DISCUSSION THREAD

**Discussion by Instantgrow** [apify/fingerprint-suite#336] [Q&A]
issue: Header is inconsistent after implement the fingerprints..
tested on: https://www.deviceinfo.me/

**### - Header from Normal Chrome with Automation Controlled By puppeteer (Without Fingerprints Injection)**

**HTTP Request Headers:**  
Accept:text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7
Accept-Encoding:gzip, deflate, br, zstd
Accept-Language:en-US,en;q=0.9
Cache-Control:max-age=0
Connection:close
Device-Memory:8
Downlink:1.65
DPR:1
ECT:4g
Host:www.deviceinfo.me
Priority:u=0, i
RTT:250
Sec-CH-Device-Memory:8
Sec-CH-DPR:1
Sec-CH-Prefers-Color-Scheme:light
Sec-CH-Prefers-Reduced-Motion:no-preference
Sec-CH-Prefers-Reduced-Transparency:no-preference
Sec-CH-UA:"Google Chrome";v="131", "Chromium";v="131", "Not_A Brand";v="24"
Sec-CH-UA-Arch:"x86"
Sec-CH-UA-Bitness:"64"
Sec-CH-UA-Full-Version:"131.0.6778.70"
Sec-CH-UA-Full-Version-List:"Google Chrome";v="131.0.6778.70", "Chromium";v="131.0.6778.70", "Not_A Brand";v="24.0.0.0"
Sec-CH-UA-Mobile:?0
Sec-CH-UA-Model:""
Sec-CH-UA-Platform:"Windows"
Sec-CH-UA-Platform-Version:"10.0.0"
Sec-CH-UA-WoW64:?0
Sec-CH-Viewport-Height:925
Sec-CH-Viewport-Width:1959
Sec-Fetch-Dest:document
Sec-Fetch-Mode:navigate
Sec-Fetch-Site:none
Sec-Fetch-User:?1
Upgrade-Insecure-Requests:1
User-Agent:Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36
Viewport-Width:1959
architecture:x86
bitness:64
brands:[{"brand":"Google Chrome","version":"131"},{"brand":"Chromium","version":"131"},{"brand":"Not_A Brand","version":"24"}]
fullVersionList:[{"brand":"Google Chrome","version":"131.0.6778.70"},{"brand":"Chromium","version":"131.0.6778.70"},{"brand":"Not_A Brand","version":"24.0.0.0"}]
mobile:false
model:
platform:Windows
platformVersion:10.0.0
uaFullVersion:131.0.6778.70
wow64:false

===========================

### **- Header from Normal Chrome with Automation Controlled By puppeteer + Fingerprints Injected**

**HTTP Request Headers:**
Accept:text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7
Accept-Encoding:gzip, deflate, br, zstd
Accept-Language:ru,en;q=0.9
Connection:close
Device-Memory:8
Downlink:1.6
DPR:1
ECT:4g
Host:www.deviceinfo.me
Priority:u=0, i
RTT:150
Sec-CH-Device-Memory:8
Sec-CH-DPR:1
Sec-CH-Prefers-Color-Scheme:light
Sec-CH-Prefers-Reduced-Motion:no-preference
Sec-CH-Prefers-Reduced-Transparency:no-preference
Sec-CH-Viewport-Height:953
Sec-CH-Viewport-Width:1920
Sec-Fetch-Dest:document
Sec-Fetch-Mode:navigate
Sec-Fetch-Site:none
Sec-Fetch-User:?1
Upgrade-Insecure-Requests:1
User-Agent:Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36
Viewport-Width:1920
brands:
mobile:
platform:
architecture:x86
bitness:64
fullVersionList:[{"brand":"Chromium","version":"130.0.6723.70"},{"brand":"Google Chrome","version":"130.0.6723.70"},{"brand":"Not?A_Brand","version":"99.0.0.0"}]
model:
platformVersion:15.0.0
uaFullVersion:130.0.6723.70
wow64:

lots of values missing after injecting the Fingerprint even with header config..

**Comment by barjin** [comment]
Hello and thank you for your interest in this project!

A lot of issues in this repo refer to some fingerprinting benchmarks that don't necessarily reflect real world antibot services. Please, report only cases where you were trying to access (scrape, automate) an actual website (Netflix, Amazon, Google Maps...) and it didn't work. While optimizing the tool for benchmarks might help with real-world usage, it also draws our attention and manpower to artificial problems.
